sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -269,6 +250,24 @@ def _monitor_process_bundle(self):
               logging.warning('Unable to schedule instruction %s for %s',
                               instruction_id, scheduling_delay)
 
+  def _get_or_create_worker(self):
+    """Gets an idle worker or creates a new one if none are idle."""
+    try:
+      self.workers.get(wait=False)
+    except queue.Empty:
```

Comment:
We're now creating a new worker any time we need work and the queue is empty. When we return the worker after processing, do we have a limit to the number allowed in the queue? If not, then the queue size will always grow to the high water mark of number needed.

Reply:
Yes, a future change can clean-up unused SDK workers.